src/OrchardCore.Modules/OrchardCore.Notifications/Drivers/NotificationNavbarDisplayDriver.cs

```diff
@@ -38,20 +38,22 @@ public override async Task<IDisplayResult> DisplayAsync(Navbar model, BuildDispl
             return null;
         }
 
-        var result = Initialize<UserNotificationNavbarViewModel>("UserNotificationNavbar", async model =>
-        {
-            var userId = _httpContextAccessor.HttpContext.User.FindFirstValue(ClaimTypes.NameIdentifier);
-            var notifications = (await _session.Query<Notification, NotificationIndex>(x => x.UserId == userId && !x.IsRead, collection: NotificationConstants.NotificationCollection)
-                .OrderByDescending(x => x.CreatedAtUtc)
-                .Take(_notificationOptions.TotalUnreadNotifications + 1)
-                .ListAsync()).ToList();
+        var result = Initialize<UserNotificationNavbarViewModel>("UserNotificationNavbar")
+            .Processing<UserNotificationNavbarViewModel>(async model =>
```

Comment:
@sebastienros if we can't get the Cache() to work on the on the ShapeResult, shouldn't we remove it then?

Also, since we are changing Initilize<TModel>().Processing() should probably already take the same TModel instead of having to pass that again. It just seems weird to me to have to pass the type multiple times in the same chain.

Reply:
> if we can't get the Cache() to work on the on the ShapeResult, shouldn't we remove it then

This works

> should probably already take the same TModel

I didn't want to introduce a breaking change, or more involved one, as it would require a new ShapeResult<T> and other related changes. This change works.

Reply:
So the caching work, but to avoid initializing on every request, we have to delegate the initialize logic to the Processing method. Correct? If so, I am not sure the API is intuitive enough. My first instinct would be to initialize as I would normally do in the call back.

It would been much better if we did not have to explain the use of Processing method in this case as it complicate the code.